src/lazy/expanded/template.rs

```diff
@@ -141,6 +141,63 @@ impl Parameter {
             ParameterCardinality::ZeroOrOne | ParameterCardinality::ZeroOrMore
         )
     }
+
+    /// If the parameter's encoding matches the provided argument, returns `Ok`.
+    /// Otherwise, returns an `Err`.
+    #[inline]
+    pub fn expect_encoding(&self, arg_encoding: &ParameterEncoding) -> IonResult<&Self> {
```

Comment:
🪧 Helper methods for validating that the current macro signature parameter has the expected encoding/cardinality.